packages/world/src/World.sol

```diff
@@ -157,6 +157,29 @@ contract World is StoreRead, IStoreData, IWorldKernel {
     StoreCore.deleteRecord(resourceSelector, key);
   }
 
+  /************************************************************************
+   *
+   *    WORLD EPHEMERAL METHODS
+   *
+   ************************************************************************/
+
+  /**
+   * Emit the ephemeral event without modifying storage at the given namespace and name.
+   * Requires the caller to have access to the namespace or name.
+   */
+  function setEphemeralRecord(
```

Comment:
I think we should move this to a module to avoid increasing the World contract's bytecode size - can even be the existing `CoreModule` (note that nothing changes for consumers since a root function selector is registered)